### PR TITLE
[aws-sdk-cpp] Update AWS SDK for C++ and its dependencies to 1.11.132

### PR DESCRIPTION
**Pull request by ConorRyan**
I noticed a memory leak in S3 so I updated all of AWS SDK while I'm at it to the most recent tagged version that contains [a fix](https://github.com/awslabs/aws-c-s3/pull/279).

I updated the dependencies based on the listed upstream ones in the `builder.json` files.

I also replaced two `aws-sdk-cpp` patches with a new option as [that is preferred](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#prefer-options-over-patching).

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by ConorRyan** [comment]
@microsoft-github-policy-service agree

File: ports/aws-sdk-cpp/fix-openssl3.patch [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-index cae5b92248..1aa728b01c 100644
---- a/cmake/compiler_settings.cmake
-+++ b/cmake/compiler_settings.cmake
-@@ -53,7 +53,7 @@ macro(set_gcc_flags)
- endmacro()
- 
- macro(set_gcc_warnings)
--    list(APPEND AWS_COMPILER_WARNINGS "-Wall" "-Werror" "-pedantic" "-Wextra")
-+    list(APPEND AWS_COMPILER_WARNINGS "-Wno-deprecated-declarations" "-Wall" "-Werror" "-pedantic" "-Wextra")
-     if(COMPILER_CLANG)
```
Comment (Adela0814) [inline ports/aws-sdk-cpp/fix-openssl3.patch:11]: Why you remove this patch? option `AWS_ SDK_ WARNINGS_ ARE_ ERRORS` doesn't seem to contain `-Wno-deprecated-declarations`.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/aws-sdk-cpp/fix-openssl3.patch (hunk 1 above)
Comment (ConorRyan) [reply ports/aws-sdk-cpp/fix-openssl3.patch:11]: You are correct, apologies for my inaccurate description I forgot about this - I fixed this patch but then thought I would check the build without, and it seems it is no longer needed thanks to https://github.com/aws/aws-sdk-cpp/pull/1986 .

The version used when this patch was written did not contain that fix: https://github.com/aws/aws-sdk-cpp/tree/1.9.160

Comment (ConorRyan) [reply ports/aws-sdk-cpp/fix-openssl3.patch:11]: Patch added [here](https://github.com/microsoft/vcpkg/pull/22878/commits/a6a953259998d113e8d69f2220e46dd8218e7c19)

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**